factory/state/stateComponents.go

```diff
@@ -122,34 +125,48 @@ func (scf *stateComponentsFactory) Create() (*stateComponents, error) {
 }
 
 func (scf *stateComponentsFactory) createStateChangesCollector() (state.StateChangesCollector, error) {
```

Comment:
state changes collector is created 2 times in this file, it should be creating only once for main accounts db and then passed to peer accounts

Reply:
not related to this PR, but please fix this here

Reply:
I think we should have a different collector for PeerAccounts that can be disabled separately from config.